### PR TITLE
Make user:report command scale

### PR DESCRIPTION
**Pull request by nickvergessen**
Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

File: core/Command/User/Report.php [hunk 1]
```diff
@@ -72,10 +80,12 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		} else {
 			$rows[] = ['No backend enabled that supports user counting', ''];
 		}
-
-		$userDirectoryCount = $this->countUserDirectories();
 		$rows[] = [' '];
-		$rows[] = ['user directories', $userDirectoryCount];
+
+		if ($total <= 500 || $input->getOption('count-dirs')) {
```
Comment (blizzz) [inline core/Command/User/Report.php:85]: preferably in a const, but ok.

File: core/Command/User/Report.php [hunk 2]
```diff
@@ -40,19 +43,24 @@ class Report extends Command {
...
+	protected function configure(): void {
 		$this
 			->setName('user:report')
-			->setDescription('shows how many users have access');
+			->setDescription('shows how many users have access')
+			->addOption(
+				'count-dirs',
+				null,
+				InputOption::VALUE_NONE,
+				'Also count the number of user directories in the database (could time out on huge installations, there for defaults to no with 500+ users)'
```
Comment (MorrisJobke) [inline core/Command/User/Report.php:61]:
```suggestion
				'Also count the number of user directories in the database (could time out on huge installations, therefore defaults to no with 500+ users)'
```

**Review by blizzz — APPROVED** [review]

**Review by MorrisJobke — APPROVED** [review]
CS not happy, but code looks good 👍

**Comment by MorrisJobke** [comment]
@nickvergessen Backport down to 20?

**Comment by nickvergessen** [comment]
I guess it's fine

**Pull request merged**